# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleMapKey.h
# Commit: 0527d227c6b6
# Full Hash: 0527d227c6b6f3d6648704a01bfd1c090b971ea7
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 2
# Description:
#   Bug 1742437 - Move module loader files into a central folder; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137532
# ==============================================================================

diff -r 967e68ed5f56 -r 0527d227c6b6 js/loader/ModuleMapKey.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/loader/ModuleMapKey.h	Fri Feb 18 12:45:52 2022 +0000
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef ModuleMapKey_h__
+#define ModuleMapKey_h__
+
+#include "nsURIHashKey.h"
+#include "nsIGlobalObject.h"
+
+namespace mozilla::dom {
+
+class ModuleMapKey : public nsURIHashKey {
+ public:
+  using KeyType = ModuleMapKey&;
+  using KeyTypePointer = const ModuleMapKey*;
+
+  ModuleMapKey() = default;
+  ModuleMapKey(const nsIURI* aURI, nsIGlobalObject* aWebExtGlobal);
+  explicit ModuleMapKey(const ModuleMapKey* aKey);
+  ModuleMapKey(ModuleMapKey&& aToMove) noexcept;
+
+  ModuleMapKey& operator=(const ModuleMapKey& aOther) = default;
+
+  KeyType GetKey() { return *this; }
+  KeyTypePointer GetKeyPointer() const { return this; }
+  static KeyTypePointer KeyToPointer(KeyType aKey) { return &aKey; }
+
+  bool KeyEquals(KeyTypePointer aOther) const;
+  static PLDHashNumber HashKey(KeyTypePointer aKey);
+
+  nsCOMPtr<nsIGlobalObject> mWebExtGlobal;
+};
+
+inline void ImplCycleCollectionUnlink(ModuleMapKey& aField) {
+  ImplCycleCollectionUnlink(aField.mWebExtGlobal);
+}
+
+inline void ImplCycleCollectionTraverse(
+    nsCycleCollectionTraversalCallback& aCallback, ModuleMapKey& aField,
+    const char* aName, uint32_t aFlags = 0) {
+  ImplCycleCollectionTraverse(aCallback, aField.mWebExtGlobal, aName, aFlags);
+}
+
+}  // namespace mozilla::dom
+
+#endif